src/ConnectionPool.jl

```diff
@@ -240,7 +262,7 @@ function IOExtras.startwrite(t::Transaction)
     @require !iswritable(t)                     ;t.c.writecount != t.sequence &&
                                                    @debug 1 "⏳  Wait write: $t"
     while t.c.writecount != t.sequence
-        wait(t.c.writedone)
+        _wait(t.c.writedone)
     end                                           ;@debug 2 "👁  Start write:$t"
     t.c.writebusy = true
```

Comment:
You almost certainly want to be holding the lock around this entire section, not just within the `wait` function itself. Note this example on wikipedia about how to use a Monitor (a combo Condition Variable + Mutex, which is what Julia's `Condition()` type provides):
https://en.wikipedia.org/wiki/Monitor_(synchronization)#Monitor_usage
```
acquire(m); // Acquire this monitor's lock.
while (!p) { // While the condition/predicate/assertion that we are waiting for is not true...
	wait(m, cv); // Wait on this monitor's lock and condition variable.
}
// ... Critical section of code goes here ...
signal(cv2); -- OR -- notifyAll(cv2); // cv2 might be the same as cv or different.
release(m); // Release this monitor's lock.
```

You'll need to lock during accesses to `writebusy`, `writecount`, `sequence`, etc, here, lest they be updated simultaneously from multiple threads.

It might be that this entire function, actually, should be within the lock.